src/main/java/com/ibm/cloudant/kafka/connect/CloudantSinkTask.java

```diff
@@ -48,65 +42,32 @@ public class CloudantSinkTask extends SinkTask {
 	private int taskNumber;
 	public static String guid_schema = null;
 	private Boolean replication;
-	public static volatile JSONArray jsonArray = new JSONArray();
 
+	private List<Map<String, Object>> jsonArray = new LinkedList<>();
+
+	private static ConnectRecordMapper<SinkRecord> mapper = new ConnectRecordMapper<>();
 	
 	@Override
 	public String version() {
 		 return new CloudantSinkConnector().version();
 	}
 
-	
 	//TODO: all sinkRecords in first Thread
 	@Override
 	public void put(Collection<SinkRecord> sinkRecords) {
-	
+
 		LOG.info("Thread[" + Thread.currentThread().getId() + "].sinkRecords = " + sinkRecords.size());
 		
-		for (SinkRecord record : sinkRecords) {		
-			JSONObject jsonRecord;
-		
-			JSONTokener tokener = new JSONTokener(record.value().toString());		
-			jsonRecord = new JSONObject(tokener);
-											
-			if (jsonRecord.has(CloudantConst.CLOUDANT_REV)) {
-				jsonRecord.remove(CloudantConst.CLOUDANT_REV);
-			}
-			
-			if(jsonRecord.has(CloudantConst.CLOUDANT_DOC_ID)){			
-				if(replication == false) {
-					//Add archive schema from SinkRecord when available
-					jsonRecord.put(InterfaceConst.KC_SCHEMA, record.valueSchema());
-					
-					//Create object id from kafka
-					jsonRecord.put(CloudantConst.CLOUDANT_DOC_ID, 
-							record.topic() + "_" + 
-							record.kafkaPartition().toString() + "_" + 
-							Long.toString(record.kafkaOffset()) + "_" + 
-							jsonRecord.get(CloudantConst.CLOUDANT_DOC_ID));	
-				}
-				//OPTION B: IF replication == true => Do Nothing => Create mirror from Cloudant object
-				
-				//OPTION C (not implemented): generate new id with  cloudant 
-				/*else {
-					LOG.info(MessageKey.GUID_SCHEMA + ": " + guid_schema);
-					LOG.warn(CloudantConst.CLOUDANT_DOC_ID + "from source database will removed");
-					
-					//remove Cloudant _id
-					jsonRecord.remove(CloudantConst.CLOUDANT_DOC_ID);
-				}*/
-			}					
-			jsonArray.put(jsonRecord);
-			
-			if ((jsonArray != null) && (jsonArray.length() >= batch_size)) {
-	
+		for (SinkRecord record : sinkRecords) {
+			Map<String, Object> recordValue = mapper.apply(record);
+			recordValue.remove(CloudantConst.CLOUDANT_REV);
+			jsonArray.add(recordValue);
+			if (jsonArray.size() >= batch_size) {
 				flush(null);
-	
 			} 
 		} 
```

Comment:
Not essential, but now that the `mapper` is a functional interface and the code around here is more streamlined we could potentially use streams to take advantage of parallelism for the "mapping" part of the loop before sequentially processing the add/flush e.g. something like:
```java
		sinkRecords.stream()
			.map(mapper) // Convert ConnectRecord to Map
			.map(recordValueAsMap -> {
				recordValueAsMap.remove(CloudantConst.CLOUDANT_REV); // Remove the _rev
				return recordValueAsMap;
			})
			.sequential() // Avoid concurrent access to jsonArray
			.forEach(recordValueAsMap -> {
				jsonArray.add(recordValueAsMap);
				if (jsonArray.size() >= batch_size) {
					flush(null);
				} 
			});
```